cds_ils/circulation/indexer.py

```diff
@@ -18,12 +19,17 @@ def index_extra_fields_for_loan(loan_dict):
     needed when fetching the loan details.
     """
     document_class = current_app_ils.document_record_cls
-    document_record = document_class.get_record_by_pid(
-        loan_dict.get("document_pid")
-    )
+    try:
+        document_record = document_class.get_record_by_pid(
+            loan_dict["document_pid"]
+        )
+    except PIDDeletedError:
+        # Document might have been deleted while reindexing asynchronously.
+        return
+
     document = document_record.replace_refs()
 
-    on_shelf = document.get("items", []).get("on_shelf", {})
+    on_shelf = document["items"]["on_shelf"]
```

Comment:
I guess that the extra below if now redundant... ping @kprzerwa to double check

Reply:
I think the if below saves us some minimal dictionary operations and it could stay. Especially because we have cases of this field being empty

but after considering this part, I have some doubts if either solution is a good practice in the context of what we do below and maybe we should copy this dict instead of assigning value. Some lines below, we pop some values while still pointing at the original record. While here it will probably have no consequences, since it is indexer of a loan, I am worried it could happen in other places where there is a real possibility of updating the record dict. What do you think @ntarocco ?

Reply:
I discussed with @ntarocco , @Pineirin could you add a comment below, before line 38 explaining the reasoning of the following operations ?

Reply:
To be 100% clear, I would add a comment explaining what `item_suggestion` is doing and why we are doing all those IFs...